activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -201,6 +205,26 @@ def disable_referential_integrity # :nodoc:
       # DATABASE STATEMENTS ======================================
       #++
 
+      def bulk_change_table(table_name, operations)
+        combinable, non_combinable = Array(operations).partition { |c, args| respond_to?(:"#{c}_for_alter", true) }
```

Comment:
I'm concerned that this is giving up the ordering guarantee we supply in the other adapters: that consecutive combinable operations will be performed together, but that any non-combinable operation will see the database in the state any preceding operations left it.

Reply:
I see what you mean about giving up the ordering guarantee. I was thinking that, especially in the case of sqlite, it would be advantageous to group all combinable operations to reduce the number of times `move_table` is executed. I tried to come up with a case where pulling column changes (add_column, add_timestamp, remove_column, change_column) up in the ordering would be problematic for later operations. I couldn't come up with any but that might just be confirmation bias on my part.

I'll rework this to be symmetric with the other adapters!